src/lbaf/Model/lbsPhase.py

```diff
@@ -406,6 +407,143 @@ def populate_from_log(self, phase_id):
         print_function_statistics(
             objects, lambda x: x.get_overhead(), "object overheads", self.__logger)
 
+    def populate_from_specification(self, spec: PhaseSpecification, multiple_sharing: bool = False):
+        """Populate this phase using a specification listing phases, ranks, tasks and communications."""
+
+        # This method is inspired by the VTDataReader but with a DatasetSpecification input
+
+        # Auto-generate ids for Lists by converting to dictionaries to set indices as keys for tasks, shared blocks,
+        # and communications
+        if isinstance(spec["tasks"], list):
+            spec["tasks"] = {i:spec["tasks"][i] for i in range(len(spec["tasks"]))}
+        if isinstance(spec["shared_blocks"], list):
+            spec["shared_blocks"] = {i:spec["shared_blocks"][i] for i in range(len(spec["shared_blocks"]))}
+        if isinstance(spec["communications"], list):
+            spec["communications"] = {i:spec["communications"][i] for i in range(len(spec["communications"]))}
+
+        ranks: Dict[int,Rank] = {r_id:Rank(self.__logger, r_id) for r_id in spec.get("ranks", []).keys()}
+        objects: Dict[int,Object] = {} # set of objects (tasks)
+        shared_blocks: Dict[int,Block] = {} # set of objects (tasks)
+
+        if len(spec["ranks"].keys()) == 0:
+            raise RuntimeError("Missing rank distributions")
+
+        # Load objects set
+        for rank_id, rank_specs in spec["ranks"].items():
+            task_ids = rank_specs["tasks"]
+            for task_id in task_ids:
+                task_user_defined = {}
+                time = spec["tasks"][task_id]
+
+                # Check: task must be only in 1 rank
+                if task_id in objects:
+                    raise RuntimeError(f"Task already in rank {objects[task_id].get_rank_id()}")
+
```

Comment:
Add comment here or remove blank line

Reply:
@ppebay Do you mean le line 441 ? (there is just one blank line there and there is a statement on the line 442.

Reply:
yes sorry I was unclear. Block of code below is a bit obscure IMO and a comment would help

Reply:
I did a change for that is more clear ? Or do you want to suggest some another text ?